John Miller Project 3 compleated

ThunderBroJohn opened the pull request:
Project 3 completed.

ThunderBroJohn committed:
John Miller Interpolation part 1

Iterpolation now working

ThunderBroJohn committed:
John Miller Project 3

Color and image interpolation work

calliday commented on definitions.h:
```diff
@@ -223,10 +229,26 @@ class BufferImage : public Buffer2D<PIXEL>
+        //image
+        double u;
+        double v;
+
```
Instead of separate values for rgb and uv, you could implement a double array of size 16. What I did is create a vector that I pushed back values to and looped through later in the interpolation.

ThunderBroJohn committed:
John Miller Project 4 Perspective Correct

Perspective correct compleated

ThunderBroJohn committed:
John Miller VertexShader

Vertex shader now functioning!
GraphicMatrix.h added
Shaders moved to shaders.h

ThunderBroJohn committed:
Pipeline complete - John Miller

full pipeline is functioning.

ThunderBroJohn committed:
ObjLoaderProject1 John Miller

Broken obj loader. Loads the object, couldn't get the display to work.

ThunderBroJohn committed:
Phong Shader (broken)

Something is broken with my vertex shader, however the shaders should be working. I am hoping to fix it by tomorrow but I am turning in what I can now until I solve it.

ThunderBroJohn committed:
Bunny works

I found the error in loading attributes. Had a disconect between some data. Should be all fixed now.
